msdk-featuredetection/msdk-targeteddetection/src/main/java/io/github/msdk/featuredetection/targeteddetection/TargetedDetectionMethod.java

```diff
@@ -164,14 +173,25 @@ public TargetedDetectionMethod(@Nonnull List<IonAnnotation> ionAnnotations,
                 chromatogram.setDataPoints(newDataPoints);
             }
 
+            // Set the m/z value for the chromatogram
+            double[] mzValues = buildingChromatogram.getMzValues();
+            double sum = 0;
+            int values = 0;
+            for(int i=0; i < buildingChromatogram.getSize() ; i++) {
+                if (mzValues[i] > 0) {
+                    values++;
+                    sum = sum + mzValues[i];
+                }
+            }
+            double newMz = sum/values;
+            chromatogram.setMz(newMz);
+
             // Add the chromatogram to the chromatogram list

```

Comment:
I think it might make sense to re-evaluate how the m/z values are calculated. The problem with a median value is:
- it doesn’t average the variation measured by the instrument
- it is sensitive to long peak tailing

Here are two solutions which could improve the m/z calculation:
- average m/z based on weighing of the intensities
- average m/z of all data points above the FWHM

I know that changing the way we calculate the m/z value will give different results compared to MZmine 2. But improving the m/z calculation should have a positive effect on:
- ppm values
- alignment of features
- identification of features when based on m/z values

What do you think?